projects/core/components/primitive-checkbox/primitive-checkbox.component.ts

```diff
@@ -67,9 +69,7 @@ export class TuiPrimitiveCheckboxComponent {
     }
 
     get icon(): PolymorpheusContent<TuiContextWithImplicit<TuiSizeL>> {
-        return this.value === null
-            ? this.options.icons.indeterminate
-            : this.options.icons.checked;
+        return this.computedIcon(this.value);
```

Comment:
Getter updates the state. Not good :(

Reply:
You're right( I moved icon setting into the value setter. What do you think about that?

Reply:
Ok, let's leave it like that, except `icon` getter is unnecessary now, just change `currentIcon` to `icon` and make it public.

Reply:
ok, done. I also added initial icon value, for smooth first animation